LLama/NativeLibraryConfig.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+
+namespace LLama
+{
+#if NET6_0_OR_GREATER
+    /// <summary>
+    /// A class about configurations when loading native libraries. 
+    /// Note that it could be configured only once before any call to llama model apis.
+    /// </summary>
+    public class NativeLibraryConfig
+    {
+        private static NativeLibraryConfig? instance;
+        private static readonly object lockObject = new object();
+
+        /// <summary>
+        /// Whether there's already a config for native library.
+        /// </summary>
+        public bool Initialied { get; private set; }
+        internal Description Desc { get; private set; }
+
+        internal static NativeLibraryConfig GetInstance()
+        {
+            if (instance is null)
+            {
+                lock (lockObject)
+                {
+                    if (instance is null)
+                    {
+                        instance = new NativeLibraryConfig();
+                    }
+                }
+            }
+            return instance;
+        }
+
+        /// <summary>
+        /// Load a specified native library as backend for LLamaSharp
+        /// </summary>
+        /// <param name="libraryPath"></param>
+        /// <exception cref="InvalidOperationException"></exception>
+        public static void WithLibrary(string libraryPath)
+        {
+            var config = GetInstance();
+            if (config.Initialied)
+            {
+                throw new InvalidOperationException("NativeLibraryConfig could be configured only once before any call to llama model apis.");
+            }
+            config.Desc = new Description(libraryPath);
+        }
+
+        /// <summary>
```

Comment:
I think it'd be better to split this into various `WithX` methods rather than having them all in one method (more extensible in the future).

e.g.

```csharp
WithCuda(bool cuda)
WithAvx(AvxLevel avx)
WithAutoFallback(bool allow)
WithSkipCheck(bool skip)
```

Reply:
This format seems good, especially for the ability of chained call. There's one problem that it couldn't notice user the default configuration well. If we just set default value to the parameter, it may looks strange when calling `WithAutoFallback().WithSkipCheck(false)`. 😹

Anyway, I prefer this format, too. However is there any solution about the case above?